src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -104,7 +124,7 @@ internal static ReportDiagnostic GetDiagnosticReport(DiagnosticSeverity severity
                 location.SourceTree != null &&
                 ((SyntaxTree)location.SourceTree).GetPragmaDirectiveWarningState(id, location.SourceSpan.Start) == ReportDiagnostic.Suppress)
             {
-                return ReportDiagnostic.Suppress;
+                hasPragmaSuppression = true;
```

Comment:
Looks like this function will return ReportDiagnostics.Default in this case (i.e. when hasPragmaSuppression = true)? Shouldn't we also return ReportDiagnostic.Suppress in this case?